python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -642,24 +799,76 @@ def get_latest_version(self, constraint_pattern=None):
             )
             if version_to_use is None:
                 raise TankDescriptorError(
-                    "'%s' does not have a version matching the pattern '%s'. "
-                    "Available versions are: %s"
-                    % (
-                        self.get_system_name(),
-                        constraint_pattern,
-                        ", ".join(version_numbers),
-                    )
+                    f"'{self.get_system_name()}' does not have a version matching the pattern '{constraint_pattern}'. "
+                    f"Available versions are: {', '.join(version_numbers)}"
                 )
             # get the sg data for the given version
             sg_data_for_version = [
                 d for d in matching_records if d["code"] == version_to_use
             ][0]
 
         else:
-            # no constraints applied. Pick first (latest) match
+            # no constraints applied. Check if latest version is Python compatible
```

Comment:
Could we make this code less "Python version" specific and more generic for any current or future "requirements" we want to have?